pkg/terminal/command.go

```diff
@@ -1341,6 +1351,175 @@ func edit(t *Term, ctx callContext, args string) error {
 	return cmd.Run()
 }
 
+func parseFmtStr(fmtStr string) (count int, format byte, size int, err error) {
+	if len(fmtStr) <= 2 {
+		return 0, ' ', 0, fmt.Errorf("invalid argument for fmt")
+	}
+	if fmtStr[0] != '-' {
+		return 0, ' ', 0, fmt.Errorf("the first letter of fmt should be \"-\"")
+	}
+	i := 1
+	for _, v := range fmtStr {
+		if unicode.IsDigit(v) {
+			break
+		}
+		i++
+	}
+	if i == 1 {
+		return 0, ' ', 0, fmt.Errorf("the \"count\" of fmt should be a number")
+	}
+	count, err = strconv.Atoi(fmtStr[1:i])
+	if err != nil {
+		return 0, ' ', 0, fmt.Errorf("strconv the \"count\" into number of fmt get %s", err)
+	}
+
+	formatSupportedArrForErr := func(arr []byte) string {
+		str := "[ "
+		for i, v := range arr {
+			str += "\"" + string(v) + "\""
+			if i != len(arr)-1 {
+				str += " "
+			}
+		}
+		str += "]"
+		return str
+	}
+
+	// if no format and size, use default argument 'xb'
+	if i == len(fmtStr) {
+		return count, 'x', 1, nil
+	}
+
+	supportedFormat := []byte{'o', 'x', 'd'}
+	format = fmtStr[i]
+	hitSupportedFlag := false
+	for _, v := range supportedFormat {
+		if v == format {
+			hitSupportedFlag = true
+			break
+		}
+	}
+	if !hitSupportedFlag {
+		return 0, ' ', 0, fmt.Errorf("the \"format\" letter should be one of %s", formatSupportedArrForErr(supportedFormat))
+	}
+
+	i++
+	// if no size, use default argument 'b'
+	if i == len(fmtStr) {
+		return count, format, 1, nil
+	}
+	supportedSize := []byte{'b', 'h', 'w', 'g'}
+	hitSupportedFlag = false
+	var index int
+	for index = 0; index < len(supportedSize); index++ {
+		if supportedSize[index] == fmtStr[i] {
+			hitSupportedFlag = true
+			break
+		}
+	}
+	if !hitSupportedFlag {
+		return 0, ' ', 0, fmt.Errorf("the \"size\" letter should be one of %s", formatSupportedArrForErr(supportedSize))
+	}
+	size = 1 << uint(index)
+
+	i++
+	// if redundant parameters, should report error
+	if i != len(fmtStr) {
+		return 0, ' ', 0, fmt.Errorf("some redundant parameters for fmt")
+	}
+	return count, format, size, nil
+}
+
+func examineMemoryCmd(t *Term, ctx callContext, args string) error {
+	v := strings.FieldsFunc(args, func(c rune) bool {
+		return c == ' '
+	})
+	if len(v) != 1 && len(v) != 3 && len(v) != 5 {
+		return fmt.Errorf("wrong arguments for input")
+	}
+	// Default value
+	format := "hex"
+	length := 1
+	addressStr := v[len(v)-1]
+
+	StringArrIndex := func(strArr []string, str string) int {
```

Comment:
In principle this is fine, however it's also going to make adding new arguments unnecessarily complicated. Something like this is just as simple and more flexible:

```
for i := 0; i < len(v); i++ {
    switch v[i] {
    case "-fmt":
         i++
         // parse argument...
    case "-len":
         i++
         // parse argument...
    default:
         if i != len(v)-1 {
             return fmt.Errorf(...)
         }
         // parse argument...
    }
}
```

similar to parseStackArgs.

Reply:
Oh, it is brilliant, thx